java-checks-test-sources/src/main/java/checks/regex/RedundantRegexAlternativesCheck.java

```diff
@@ -68,7 +68,12 @@ public void f(Pattern pattern) {
     f(compile(".*|a")); // Noncompliant
     f(compile("[ab]"));
     f(compile(".*"));
-    f(compile("[\uD83D\uDE02]" + "[\uD83D\uDE0A]")); // Compliant 
-  }
+    f(compile("[\uD83D\uDE02]" + "[\uD83D\uDE0A]")); // Compliant
 
+    // POSIX character classes are not handled
+    compile("\\p{Space}|x"); // Compliant
+    compile("x|\\p{Space}"); // Compliant
+    compile("\\p{Space}| "); // Compliant - False negative
+    compile("\\p{Lower}|a"); // Compliant - False negative
+  }
```

Comment:
The "old" implementation was already not supporting POSIX character classes. I created the following issue on sonar-commons: https://github.com/SonarSource/sonar-analyzer-commons/issues/223